src/events/IdentifyProjectOnCommit.ts

```diff
@@ -0,0 +1,73 @@
+import {
+    EventFired,
+    EventHandler,
+    failure,
+    GraphQL,
+    HandleEvent,
+    HandlerContext,
+    HandlerResult,
+    logger, MappedParameter, MappedParameters, Secret, Secrets,
+    Success,
+    success,
+    Tags,
+} from "@atomist/automation-client";
+import axios from "axios";
+
+import {GitHubRepoRef} from "@atomist/automation-client/operations/common/GitHubRepoRef";
+import {GitCommandGitProject} from "@atomist/automation-client/project/git/GitCommandGitProject";
+import * as graphql from "../typings/types";
+import {ProjectProperties, ProjectPropertiesExtractor} from "../util/ProjectPropertiesExtractor";
+
+@EventHandler("fingerprint a commit with project identification",
+    GraphQL.subscriptionFromFile("../graphql/commitWithProjectIdFingerprint", __dirname))
+@Tags("blackDuck", "status")
+export class IdentifyProjectOnCommit implements
+    HandleEvent<graphql.CommitWithProjectIdFingerprint.Subscription> {
+
+    @Secret(Secrets.userToken("repo"))
```

Comment:
I don't think user tokens are available in event handlers. You have to use the orgToken.